### DISCUSSION TITLE
Backtest entry price is not matched to real price.

### DISCUSSION THREAD

**Discussion by sfolaxsel** [conor19w/Binance-Futures-Trading-Bot#37] [General]
![bb1](https://github.com/conor19w/Binance-Futures-Trading-Bot/assets/134991324/79933abd-250b-4a9f-be46-266fb37c3c50)
![bb2](https://github.com/conor19w/Binance-Futures-Trading-Bot/assets/134991324/6a988a1e-0b91-4020-a202-9dae7b101ddc)

Hello. I conducted a backtest using the ETHUSDT pair on a 15-minute timeframe. As you can see in the attached image, the bot entered a position at 1806.45 on 05-26, 2023, at 00:59:00. It indicates that the current price at that moment was 1804.65.

I wanted to compare it with the actual data, so I looked for the chart. The highest price in the timeframe from 00:45:00 to 01:00:00 on May 26, 2023, was 1792.06. And the highest price in the following timeframe was 1794.51. There is a significant difference between the values in the backtest and the actual data. I'm wondering what went wrong.

**Comment by conor19w** [comment]
Might be due to the slippage you are using, could be an error either would need to see what config you ran the backtest with

**Reply by sfolaxsel** [reply]
I tried setting slippage to 0 and testing it, but the entry price and current price still have a difference. 
Also, regardless of my config, the price at that time (probably the current price?) should be the same as the actual price, shouldn't it?
And Here is my config settings
Start 01-05-23
End 30-05-23
Starting Balance 100, Order Size 20
Candle Length 15m, 
leverage 20
SL 5 TP 3
Symbo : ETHUSDT

and strategy is my custom in StochRSIMACD

  # long
    if (
        (
            fastk[current_index] <= 10 and
            fastk[current_index] >= fastd[current_index]
        )
    ):
        Trade_Direction = 1
    # short
    elif (
        (
            fastk[current_index] >= 90 and
            fastk[current_index] <= fastd[current_index]
        )
    ):
        Trade_Direction = 0

    return Trade_Direction

**Reply by conor19w** [reply]
Have you adjusted the time to be your time zone? 
time_delta variable at the top of the scripts
Also you're using trading view I see, are you viewing the charts pulled from binance?

**Reply by sfolaxsel** [reply]
Yes, I have tried changing the time zone to my location and also tried setting it to 0, but the difference still remains significant. To make it easier for you to reproduce, I have reset the repository to the latest branch and configured the settings as follows: the symbol is only ETHUSDT. 

![image](https://github.com/conor19w/Binance-Futures-Trading-Bot/assets/134991324/fa0f11c2-8958-406c-b14c-0418fb6b3e3b)

After that, when I click the Run BackTest button, the last trade result is displayed as follows, with 1890.8 being a price that does not exist during that time period. And that chart is from Binance.

![image](https://github.com/conor19w/Binance-Futures-Trading-Bot/assets/134991324/2e104ffe-68da-40d9-bc80-8283fb76236e)

![image](https://github.com/conor19w/Binance-Futures-Trading-Bot/assets/134991324/2684972c-1f60-4ac8-a502-931c5e196378)

Is there a feature in the backtesting functionality that allows inputting a date and time to fetch the price at that specific time? I would like to verify if the backtest is being performed with accurate data. Thanks!

**Comment by conor19w** [comment]
I checked the price in your first image the price matches that of binance (the time 00:59 is the opening time of the candle on the 1m timeframe so this is 01:00) see current price 1804.65 in the console matches the open of this image and then possibly slippage effected your entry price:
<img width="901" alt="image" src="https://github.com/conor19w/Binance-Futures-Trading-Bot/assets/63513457/18f26e00-db97-4226-b21f-3f3a6ebee945">

**Reply by sfolaxsel** [reply]
ahh okay. i'll check slippage again. thank you!

**Reply by sfolaxsel** [reply]
I have adjusted the timedelta based on the 'UTC+x' displayed at the bottom of the Binance chart and confirmed that the current price matches.  and set slippage to 0 makes correct entry price. Thank you!
